client/src/pages/ZipForecast.js

```diff
@@ -1,7 +1,41 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { fetchForecast } from "../api/fetchForecast";
+import { useParams } from "react-router-dom";
+import LoadingScreen from "../components/LoadingScreen";
+import ErrorScreen from "../components/ErrorScreen";
 
 const ZipForecast = () => {
-  return <div>forecast charts</div>;
+  const { zip } = useParams();
+  const [forecastData, setForecastData] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        setLoading(true);
+        setError(null);
+        const fetchedForecast = await fetchForecast(zip);
+        setForecastData(fetchedForecast);
+        setLoading(false);
+      } catch (error) {
+        console.error(error);
+      }
```

Comment:
Don't forget to set the error state.

And reset `loading` (maybe in `finally`)